src/github.com/matrix-org/dendrite/clientapi/routing/joinroom.go

```diff
@@ -272,6 +276,12 @@ func (r joinRoomReq) joinRoomUsingServers(
 // server was invalid this returns an error.
 // Otherwise this returns a JSONResponse.
 func (r joinRoomReq) joinRoomUsingServer(roomID string, server gomatrixserverlib.ServerName) (*util.JSONResponse, error) {
+	// parse all data (and return an error) before doing some work
+	evTime, resErr := httputil.ParseTSParam(r.req)
+	if resErr != nil {
+		// this looks weird here but does what we want.
+		return resErr, resErr.JSON.(jsonerror.MatrixError)
```

Comment:
The error here is not due to communication with the other server, hence
`resErr, nil` makes more sense.

Reply:
I expect calling functions to check for the response of the `error` param to identify if there is an issue. This structure returns the prase which the creator of `resErr` intended to have there